src/services/formatting/formatting.ts

```diff
@@ -78,7 +78,7 @@ namespace ts.formatting {
         //  1. the end of the previous line
         //  2. the last non-whitespace character in the current line
         let endOfFormatSpan = getEndLinePosition(line, sourceFile);

```

Comment:
`isWhiteSpace` will already return false on line breaks, so when we reach the end of the previous line the `while` loop will terminate.